### PR TITLE
Vanadium flow battery: voltage model fixes

### PR DESCRIPTION
**Pull request by dguittet**
Fixes:
 - Copied over some fixes from `develop` on `calculate_max_charge_w` and `calculate_max_discharge_w`
 - fix `-inf` potential from `voltage_model`
 - voltage drop across aggregated stack the same for charge or discharge
 - adapt test to show catch difference in using abs(I_string)

With the thermal model turned off: this is the yearly battery charging & discharging table:
image

### CODE REVIEW

File: shared/lib_battery.cpp [hunk 1]
```diff
@@ -925,22 +928,25 @@ double voltage_vanadium_redox_t::voltage_model(double q0, double qmax, double I_
 	double SOC_use = q0 / qmax;
 	if (SOC_use > 1. - tolerance)
 		SOC_use = 1. - tolerance;
+	if (SOC_use == 0)
+	    SOC_use = 1e-3;
 
 	double A = std::log(std::pow(SOC_use, 2) / std::pow(1 - SOC_use, 2));
-
-	return _V_ref_50 + m_RCF * T * A + I_string * _R;
+	return _V_ref_50 + m_RCF * T * A + fabs(I_string) * _R;
```
Comment (brtietz) [inline shared/lib_battery.cpp:935]: Can you add a comment here about how our voltage assumptions are different from the paper? (justifying the fabs call)

Comment (dguittet) [reply shared/lib_battery.cpp:935]: Sure, good point!

Comment (dguittet) [reply shared/lib_battery.cpp:937]: // In constrast to the V_stack + I_stack * R_specific in the paper which follows the convention of negative voltages,
// here the abs(I_stack) is used to allow both terms to move in same direction (https://github.com/NREL/ssc/issues/404)

**Comment by dguittet** [comment]
@cpaulgilman Yes, it would be a separate PR. It think it's easier to review and track changes when the Pull requests (and Issues) are smaller and more targeted.

I haven't gotten a chance to look at the Li-ion part, but I'll move it to an issue of its own about the voltage table.

**Pull request merged**